security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -648,9 +649,18 @@ func (sc *SecretManagerClient) generateNewSecret(resourceName string) (*security
 	}, nil
 }
 
-var rotateTime = func(secret security.SecretItem, graceRatio float64) time.Duration {
+var rotateTime = func(secret security.SecretItem, graceRatio float64, graceRatioJitter float64) time.Duration {
+	// stagger rotation times to prevent large fleets of clients from renewing at the same moment.
+	jitter := (rand.Float64() * graceRatioJitter) * float64(rand.IntN(3)-1)
```

Comment:
Linter isn't happy about math/rand/v2. What do ya'll think? Do we care about using `crypo/rand` for this purpose? I can update if so, or just put `// # nosec G402 - crt randomness is not worth the cost here` on this line.
```

Reply:
I think just opting out here is fine